ludwig/models/predictor.py

```diff
@@ -194,9 +209,14 @@ def batch_evaluation(self, dataset, collect_predictions=False, dataset_name=None
                     if collect_predictions:
                         for of_name, of_preds in preds.items():
                             for pred_name, pred_values in of_preds.items():
-                                if pred_name not in EXCLUDE_PRED_SET:
+                                if collect_logits or pred_name not in EXCLUDE_PRED_SET:
                                     key = f"{of_name}_{pred_name}"
                                     predictions[key].append(pred_values)
+                    # accumulate labels from batch for each output feature
+                    if collect_labels:
+                        for of_name, of_labels in targets.items():
+                            key = f"{of_name}_labels"
+                            predictions[key].append(of_labels)
```

Comment:
I understand why this is needed, but there's an alternative. Outside of this function you could collect the labels by appending the labels column from the dataset to the output of this function. it doesn't need to live inside this predict function imho, but it could live after line 44-46 in calibrator.py

Reply:
I'm not sure how to get the labels column without: initializing a batcher, reading through the dataset batch-by-batch, and accumulating the prediction columns.  Seemed to me that tacking this onto the predict loop in Predictor would be better code re-use, and not add much cost to batch_evaluation.

Is there a faster way to just get the feature's label column from dataset?

Reply:
Aha, I figured out how to do this.